frappe/utils/nestedset.py

```diff
@@ -148,13 +151,12 @@ def rebuild_tree(doctype, parent_field):
 	right = 1
 	table = DocType(doctype)
 	column = getattr(table, parent_field)
-
 	result = (
 		frappe.qb.from_(table)
 		.where(
 			(column == "") | (column.isnull())
 		)
-		.orderby(table.name, order=Order.asc)
+		.orderby(Replace(table.name, '_', ''), order=Order.asc)
```

Comment:
Postgres sorts underscores differently than maria due to collation differences. Anything with underscore in the front gives different results. This normalizes that result.

Reply:
We could normalize the result at a lower level - perhaps in `frappe.qb`, `DatabaseQuery`, or somewhere similar.

cc: @saxenabhishek @Aradhya-Tripathi @surajshetty3416

Reply:
FWIW, it's not just underscores that this happens with. Dashes, for example, are treated differently as well. This just happens to be the only place in Frappe that it happened - but in ERPNext PR there is at least one other place where dash sorting also mattered

Reply:
@cpdeethree Did you add the `Replace` bits to fix a failing test? Because if so, I say we rather fix the tests instead. 

---

Normalizing the result like this for random queries will lead to inconsistencies in the system which will make debugging harder. Also, seems more expensive...we could try recommending similar database configurations. This shouldn't be fixed in the application IMO.

Reply:
Alright, I can back this out and use a different set of companies for this specific test suite